### PR TITLE
AI query build improvements

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
Make the following improvements to the AI Query builder
- Remove "Edit query" option
- Show prompt to users to enable Harold AI query builder

https://www.loom.com/share/bf2598815f184be88dc24bedef0b49d4?sid=f7401917-96ac-434b-9e03-80f685e4b887

## How did you test this change?
1) Load the logs page with setting disabled
- [ ] Option links you to settings to enable (c.c. @jay-khatri may be related to your changes)
2) Enable setting and reload logs
3) Click on the AI option
- [ ] Able to insert query
- [ ] Able to submit query with correct dates

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
N/A

### CODE REVIEW

File: frontend/src/components/Search/SearchForm/SearchForm.tsx [hunk 1]
```diff
@@ -509,12 +550,6 @@ export const Search: React.FC<{
 		startDate,
 	])
 
-	useEffect(() => {
-		// necessary to update the combobox with the URL state
```
Comment (SpennyNDaJets) [inline frontend/src/components/Search/SearchForm/SearchForm.tsx:513]: Don't believe this is necessary with the new search context

**Comment by SpennyNDaJets** [comment]
Tagged @ccschmitz directly to make sure no regressions/code smells related to search component or context

File: frontend/src/components/Search/SearchContext.tsx [hunk 2]
```diff
@@ -104,6 +114,39 @@ export const SearchContext: React.FC<Props> = ({
 	const { queryParts, tokens } = parseSearch(query)
 	const tokenGroups = buildTokenGroups(tokens)
 
+	const [pendingStartDate, setPendingStartDate] = useState<Date>()
+	const [pendingEndDate, setPendingEndDate] = useState<Date>()
+	const [pendingSelectedPreset, setPendingSelectedPreset] =
+		useState<DateRangePreset>()
```
Comment (SpennyNDaJets) [inline frontend/src/components/Search/SearchContext.tsx:120]: Copy of date options to allow editing without submitting search

File: frontend/src/pages/LogsPage/LogsPage.tsx [hunk 3]
```diff
@@ -307,7 +303,11 @@ const LogsPageInner = ({ timeMode, logCursor, presetDefault }: Props) => {
 						timeMode={timeMode}
 						savedSegmentType={SavedSegmentEntityType.Log}
 						textAreaRef={textAreaRef}
-						enableAIMode={enableAiQueryBuilder}
+						enableAIMode={
+							workspaceSettings?.workspaceSettings
+								?.ai_query_builder
+						}
+						aiSupportedSearch={aiQueryBuilderFlag}
```
Comment (SpennyNDaJets) [inline frontend/src/pages/LogsPage/LogsPage.tsx:310]: 2 options:
1. Search supports the AI query build
2. The query builder is enabled in the user's workspace

File: frontend/src/components/Search/SearchForm/SearchForm.tsx [hunk 4]
```diff
@@ -138,17 +140,38 @@ const SearchForm: React.FC<SearchFormProps> = ({
...
 		onSubmit(updatedQuery)
 	}
 
+	const handleDatesChange = (
+		start: Date,
+		end: Date,
+		preset?: DateRangePreset,
+	) => {
+		onDatesChange(start, end, preset)
+		onSubmit(query)
```
Comment (SpennyNDaJets) [inline frontend/src/components/Search/SearchForm/SearchForm.tsx:170]: Now have to submit dates and query as there can be pending changes

**Comment by jay-khatri** [comment]
Looks great. My only comment is: can we keep the original copy in the search box, even if the setting is disabled? I worry that someone might not understand what they are enabling with the new copy.

image

**Review by jay-khatri — APPROVED** [review]

File: frontend/src/components/Search/SearchContext.tsx (hunk 2 above)
Comment (ccschmitz) [reply frontend/src/components/Search/SearchContext.tsx:120]: Do we feel strongly about the UX of allowing editing the query before submitting? Feels like we could potentially simplify things in the code + UI by just submitting it 🤔 

cc @jay-khatri @julian-highlight

File: frontend/src/components/Search/SearchForm/AiSearch/index.tsx [hunk 5]
```diff
@@ -75,8 +82,19 @@ export const AiSearch: React.FC<any> = ({}) => {
...
+
+	const setSubmittedQuery = () => {
+		if (aiSuggestion) {
+			setQuery(aiSuggestion.query)
+			setPendingDates(
+				dateSuggestion.startDate,
+				dateSuggestion.endDate,
+				dateSuggestion.selectedPreset,
+			)
+			exitAiMode()
```
Comment (ccschmitz) [inline frontend/src/components/Search/SearchForm/AiSearch/index.tsx:97]: When I submitted the query it cleared the filters on me. Editing worked fine.

https://www.loom.com/share/53be6dbfd8b94dbd8709b621b3d50756?from_recorder=1&focus_title=1

File: frontend/src/components/Search/SearchForm/SearchForm.tsx (hunk 1 above)
Comment (ccschmitz) [reply frontend/src/components/Search/SearchForm/SearchForm.tsx:513]: I think you're right. Seems to work fine on page reload 👍

**Review by ccschmitz — CHANGES_REQUESTED** [review]
One bug I noticed with the query not submitting correctly that I think needs to be addressed before merging. Also had a question on the UX of editing that we might need @jay-khatri and/or @julian-highlight to chime in on.

File: frontend/src/components/Search/SearchContext.tsx (hunk 2 above)
Comment (jay-khatri) [reply frontend/src/components/Search/SearchContext.tsx:120]: Don't feel too strongly. If it simplifies things, let's remove the edit option! Wdyt @SpennyNDaJets?

Comment (SpennyNDaJets) [reply frontend/src/components/Search/SearchContext.tsx:120]: Sure thing!

File: frontend/src/components/Search/SearchForm/SearchForm.tsx (hunk 1 above)
Comment (SpennyNDaJets) [reply frontend/src/components/Search/SearchForm/SearchForm.tsx:513]: Added back since was causing the bug above

**Review by ccschmitz — APPROVED** [review]
Lookin' good! A few minor comments I noticed in the latest click test, but probably not really related to these changes and could be looked at in a follow up.

https://www.loom.com/share/39f5ee1c851247c98ebc6ae7555a27a5?from_recorder=1&focus_title=1

**Pull request merged**